odl/solvers/functional/functional.py

```diff
@@ -1020,11 +1033,11 @@ def __init__(self, func, quadratic_coeff=0, linear_term=None,
             raise ValueError(
                 "Complex-valued `constant` coefficient is not supported.")
         self.__constant = constant.real
-
+        #HELP: Need to check for range? The + should take care of it, right?
```

Comment:
Since `FunctionalQuadraticPerturb` inherits from `Functional` the `super` call is going to execute the code in `Functional.__init__`. So if the test there is sufficient (which I think it is), then this should be fine.

Reply:
It has been since forever that I worked with subclasses... 😅 (Which is probably a good indicator for my terrible style of programming)